### ISSUE TITLE
better handling of port in use

### ISSUE THREAD

**Issue by neuromusic** [chanzuckerberg/cellxgene#698]
If the 5005 port is in use, cellxgene throws an error which doesn't make it clear to users where the problem is...

```
(cellxgene) hal-9000:cellxgene-explore justinkiggins$ cellxgene launch -l umap osmFISH_SScortex_mouse_all_cells.h5ad
[cellxgene] Starting the CLI...
[cellxgene] Loading data from osmFISH_SScortex_mouse_all_cells.h5ad, this may take awhile...
[cellxgene] Warning: Annotation X will be converted to 32 bit float and may lose precision. 
[cellxgene] Warning: Annotation Y will be converted to 32 bit float and may lose precision. 
[cellxgene] Warning: Annotation _tSNE_1 will be converted to 32 bit float and may lose precision. 
[cellxgene] Warning: Annotation _tSNE_2 will be converted to 32 bit float and may lose precision. 
[cellxgene] Warning: Annotation size_pix will be converted to 32 bit float and may lose precision. 
[cellxgene] Warning: Annotation size_um2 will be converted to 32 bit float and may lose precision. 
[cellxgene] Launching! Please go to http://127.0.0.1:5005 in your browser.
[cellxgene] Type CTRL-C at any time to exit.
*OSError: [Errno 48] Address already in use*
```

Possible fixes:

### 1) improve error handling

handle error and tell the user to try another port

### 2) try other ports

automatically increment the port until an available port is found. jupyter implements this nicely, with a `--port-retries` argument to cap the number of attempts:

```
--port-retries=<Int> (NotebookApp.port_retries)
    Default: 50
    The number of additional ports to try if the specified port is not
    available.
```

**Comment by neuromusic** [comment]
cc @csweaver since the second fix is relevant to #687

**Comment by neuromusic** [comment]
EDIT: "Suggested fixes" -> "possible fixes"
